src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -975,7 +975,7 @@ internal static bool HasImplicitConstantExpressionConversion(BoundExpression sou
         {
             var constantValue = source.ConstantValue;
 
-            if (constantValue == null)
+            if (constantValue == null || constantValue.IsDefaultLiteral)
```

Comment:
Perhaps change `constantValue.IsDefaultLiteral` to `(object)source.Type == null`?